### PR TITLE
🐛fix lightbox thumbnail api bugs

### PR DESCRIPTION
**Pull request by cathyxz**
1. Be more forgiving if user puts a `lightbox-thumbnail-id` of something that doesn't exist. Fallback to normal thumbnail logic. 
2. Refactor logic for determining whether the current lightbox-ed target has a source element in the viewport to transition to. You should transition the current lightboxed element if: 
  * its source element is in the viewport
  * it is the child of an `<amp-carousel>` that is currently in the viewport
3. Refactor logic for sync-ing source carousels. There are way too many exceptions (lightbox-exclude, additional elements shoved into the same lightbox group) to logic this out nicely, so we just traverse the dom to figure out which carousel the target element is a child of, and then figure out which slide it is on.

### CODE REVIEW

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 1]
```diff
@@ -758,16 +765,43 @@ export class AmpLightboxGallery extends AMP.BaseElement {
...
    */
   shouldExit_() {
-    const element = this.getCurrentElement_().sourceElement;
-    return this.shouldAnimate_(element)
+    const target = this.getCurrentElement_().sourceElement;
+    if (!this.transitionTargetIsInViewport_(target)) {
+      return false;
+    }
+    return this.shouldAnimate_(target)
         .then(shouldAnimate => {
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: nit: This `then` clause is unnecessary.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 2]
```diff
@@ -758,16 +765,43 @@ export class AmpLightboxGallery extends AMP.BaseElement {
    * @private
    */
   shouldExit_() {
-    const element = this.getCurrentElement_().sourceElement;
-    return this.shouldAnimate_(element)
+    const target = this.getCurrentElement_().sourceElement;
+    if (!this.transitionTargetIsInViewport_(target)) {
+      return false;
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: This should return `Promise.resolve(false)` instead

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 3]
```diff
@@ -758,16 +765,43 @@ export class AmpLightboxGallery extends AMP.BaseElement {
...
         });
   }
 
+  /**
+   *
+   * @param {!Element} target
+   * @returns {boolean}
+   * @private
+   */
+  transitionTargetIsInViewport_(target) {
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: This method feels a little weird where 3/4 paths have no side effects and then 1 path has a side effect.

Comment (cathyxz) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: Yeah this deserves a bit of a refactor.

**Comment by cathyxz** [comment]
Merging master to fix merge conflicts.

**Pull request merged**